woocommerce/woocommerce-android
# Display components details

Closes: #8405

### Why
We are adding Composite Products read-only support to the Woo App.

### Description
This PR completes the support to see the product's components information in the product details screen by adding the remaining screen: Component details. Now when tapping on a product component, the app will navigate to the component detail screen where a user can consult the component options (products or categories that conform to the component) and component details (image, name, and description)

### Testing instructions

#### Prerequisites

1. Install and activate the [Composite Products](https://woocommerce.com/products/composite-products/) extension on your store.
2. Create at least one product with the Composite product type, with at least one component.

#### To test

1. Build and run the app.
2. Open the Products tab and select a composite product.
3. Tap the Components row in the product details. 
4. Confirm the app navigates to the Components list.
5. Tap on a component from the Components list.
6. Check that the app navigates to the component details screen
7. Check that the info displayed in the component details & component options is correct

### Images/gif

https://user-images.githubusercontent.com/18119390/231900757-a846d657-61a9-47cc-8016-52891fc5ece4.mp4
